task2/interface_manager.py

```diff
@@ -0,0 +1,89 @@
+import json
+from typing import List, Dict
+
+import requests
+from marshmallow import EXCLUDE
+
+from argument_parser import arg_pars
+from data_type import Interface
+from utils import get_logger
+
+# Argument parser
+args = arg_pars()
+
+# Logger
+logger = get_logger(log_path=args.log_path, log_name=args.log_name, log_level=args.log_level)
+
+
+def protocol(ssl):
+    if ssl:
+        return 'https'
+    else:
+        return 'http'
+
+
+def url_check(url):
+    try:
+        r = requests.get(url)
+        r.raise_for_status()
+    except Exception as e:
+        raise SystemExit(e)
+
+
+# Setting URL
+set_url = f'{protocol(args.ssl)}://{args.host}:{args.port}/get-all-interfaces'
+url_check(set_url)
+
+
+class InterfaceManager:
+    def __init__(self, api_url=set_url):
+        self.api_url = api_url
+        self.interface_schema = Interface.schema()
+        self.parsed_data = self._get_all_interfaces()
+
+    def _get_all_interfaces(self) -> List[Interface]:
+        response = requests.get(self.api_url)
+        parsed_interfaces = self.interface_schema.loads(json_data=response.text, many=True, unknown=EXCLUDE)
+        return parsed_interfaces
+
+    def analyze(self) -> Dict:
+        enabled_count = 0
+        types = {}
+        missing_ipv4, missing_ipv6 = [], []
+        for data in self.parsed_data:
+            if not data.ip_address_1.address:
+                missing_ipv4.append(data.name)
+            if not data.ip_address_2.address:
+                missing_ipv6.append(data.name)
+            if data.enabled:
+                enabled_count += 1
+            types[data.type] = types.get(data.type, 0) + 1
```

Comment:
@patrikprovaznik I like this but try and use defaultdict from collections library..